src/Illuminate/Support/Collection.php

```diff
@@ -54,6 +54,21 @@ public function all()
     }
 
     /**
+     * Get the average value of a given key.
+     *
+     * @param  string|null  $key
+     * @return mixed
+     */
+    public function avg($key = null)
+    {
+        if ($this->count() == 0) {
```

Comment:
Why not just:

        if ($count = $this->count()) {
            return $this->sum($key) / $count;
        }

We could avoid to call count function twice